lib/commons/dom/is-skip-link.js

```diff
@@ -13,22 +16,41 @@ const isInternalLinkRegex = /^\/?#[^/!]/;
  * @return {Boolean}
  */
 function isSkipLink(element) {
-  if (!isInternalLinkRegex.test(element.getAttribute('href'))) {
-    return false;
-  }
-
   let firstPageLink;
   if (typeof cache.get('firstPageLink') !== 'undefined') {
     firstPageLink = cache.get('firstPageLink');
   } else {
-    // define a skip link as any anchor element whose href starts with `#...`
-    // and which precedes the first anchor element whose href doesn't start
-    // with  `#...` (that is, a link to a page)
+    // define a skip link as any anchor element whose resolved href
+    // resolves to the current page and uses a fragment identifier (#)
+    // and which precedes the first anchor element whose resolved href
+    // does not resolve to the current page or that doesn't use a
+    // fragment identifier
+    const currentPage = window.location.origin + window.location.pathname;
```

Comment:
I thought we couldn't count on window.location existing. Won't this throw in JSDOM?

Reply:
`window.origin` doesn't exist in IE11, so we can't use that, but `window.location` exists in IE11 and JSDOM. It's just that JSDOM doesn't have any data for the location properties so we can never determine if `path/of/url#id` is a skip link or not.

Reply:
OK, so I think what we do is just feature test. If we can do this check, we do it. If not (like in JSDOM) we fall back to the current behavior.